src/plugins/intel_cpu/src/nodes/mkldnn_mvn_node.cpp

```diff
@@ -796,92 +922,89 @@ void MKLDNNMVNNode::prepareParams() {
 
     setPostOps(attr, true);
 
+    auto jcp = jit_mvn_config_params();
     if (mayiuse(cpu::x64::sse41)) {
         auto selectedPD = getSelectedPrimitiveDescriptor();
-        auto jcp = jit_mvn_config_params();
         jcp.src_prc = selectedPD->getConfig().inConfs[0].desc->getPrecision();
         jcp.dst_prc = selectedPD->getConfig().outConfs[0].desc->getPrecision();
         jcp.src_data_size = MKLDNNExtensionUtils::sizeOfDataType(MKLDNNExtensionUtils::IEPrecisionToDataType(jcp.src_prc));
         jcp.dst_data_size = MKLDNNExtensionUtils::sizeOfDataType(MKLDNNExtensionUtils::IEPrecisionToDataType(jcp.dst_prc));
         jcp.planar_layout = selectedPD->getConfig().inConfs[0].desc->hasLayoutType(LayoutType::ncsp);
-        jcp.normalize_variance = normalizeVariance_;
-        jcp.across_channels = execAcrossChannels_;
+        jcp.normalize_variance = mvnAttrs.normalizeVariance_;
+        jcp.across_channels = mvnAttrs.execAcrossChannels_;
         int N = 0;
-        std::tie(N, jcp.C, jcp.D, jcp.H, jcp.W) = shape5D;
-
-        if (mayiuse(cpu::x64::avx512_common)) {
-            mvn_kernel.reset(new jit_uni_mvn_kernel_f32<cpu::x64::avx512_common>(jcp, *attr.get()));
-
-            jcp.normalize_variance = false;
-            mvn_mean_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::avx512_common>(jcp));
-            if (normalizeVariance_) {
-                jcp.normalize_variance = true;
-                mvn_variance_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::avx512_common>(jcp));
-            }
-        } else if (mayiuse(cpu::x64::avx2)) {
-            mvn_kernel.reset(new jit_uni_mvn_kernel_f32<cpu::x64::avx2>(jcp, *attr.get()));
-
-            jcp.normalize_variance = false;
-            mvn_mean_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::avx2>(jcp));
-            if (normalizeVariance_) {
-                jcp.normalize_variance = true;
-                mvn_variance_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::avx2>(jcp));
-            }
-        } else if (mayiuse(cpu::x64::sse41)) {
-            mvn_kernel.reset(new jit_uni_mvn_kernel_f32<cpu::x64::sse41>(jcp, *attr.get()));
+        std::tie(N, jcp.C, jcp.D, jcp.H, jcp.W) = mvnAttrs.shape5D;
+        mvnAttrs.is_nhwc = getParentEdgeAt(0)->getMemory().getDesc().hasLayoutType(LayoutType::nspc);
+    }
 
-            jcp.normalize_variance = false;
-            mvn_mean_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::sse41>(jcp));
-            if (normalizeVariance_) {
-                jcp.normalize_variance = true;
-                mvn_variance_kernel.reset(new jit_uni_mvn_mean_variance_kernel_f32<cpu::x64::sse41>(jcp));
-            }
+    MVNKey key = {mvnAttrs, jcp, attr};
```

Comment:
let's remove class member attr, and init a temp attr in key, setPostOps for this temp one, extract data, create kernel with this temp. Then this temp attr EOL. See latest interpolate for example.

Reply:
Done. Thanks Chenhu!